kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/AsyncTransformer.java

```diff
@@ -0,0 +1,96 @@
+package org.hypertrace.core.kafkastreams.framework;
+
+import com.google.common.util.concurrent.RateLimiter;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Transformer;
+import org.apache.kafka.streams.processor.ProcessorContext;
+
+@Slf4j
+@SuppressWarnings("UnstableApiUsage")
+public abstract class AsyncTransformer<K, V, KOUT, VOUT>
```

Comment:
From offline conversation -
One thing that I think we should consider is is that to the child class, it’s not really async - they’re implementing sync same as always, they are just forced parent class now. In other words, inheritance isn't a great fit here, since the parent's purpose is just a wrapper around an implementation that should work identically. It would make more sense to compose it, kind of like how guava’s cacheloader works.

Then, existing transformers can just be wrapped like `AsyncTransformer.of(existingSyncTransformer, someAsyncConfig)`. Changes to use this would apply in the topology only, without having to touch any transform impls.

Reply:
@arun-dhingra : please sync with Aaron on this and handle it in next set of changes.